src/test/java/net/minecraftforge/test/WorldCapabilityTimerTest.java

```diff
@@ -0,0 +1,185 @@
+package net.minecraftforge.test;
+
+import java.util.Map;
+import java.util.Set;
+
+import com.google.common.collect.Maps;
+
+import net.minecraft.nbt.NBTBase;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.storage.WorldInfo;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.Capability.IStorage;
+import net.minecraftforge.common.capabilities.CapabilityInject;
+import net.minecraftforge.common.capabilities.CapabilityManager;
+import net.minecraftforge.common.capabilities.ICapabilitySerializable;
+import net.minecraftforge.event.AttachCapabilitiesEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.common.Mod.EventHandler;
+import net.minecraftforge.fml.common.event.FMLInitializationEvent;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.fml.common.gameevent.TickEvent;
+
+/** Simple mod to test world capability, by adding timer logic and check for raining time. */
+@Mod(modid=WorldCapabilityTimerTest.MODID, name="World Periodic Check Test", version="0.0.0")
+public class WorldCapabilityTimerTest {
+    public static final String MODID = "worldperiodicchecktest";
+
+    @CapabilityInject(IWorldTimer.class)
+    public static final Capability<IWorldTimer> TIMER_CAP = null;
+
+    @EventHandler
+    public void init(FMLInitializationEvent event)
+    {
+        CapabilityManager.INSTANCE.register(IWorldTimer.class, new TimerStorage(), DefaultTimer.class);
+        MinecraftForge.EVENT_BUS.register(new NormalEventHandler());
+    }
+
+    public static class NormalEventHandler {
+        @SubscribeEvent
+        public void attatchTimer(AttachCapabilitiesEvent.World event)
+        {
+            event.addCapability(new ResourceLocation(MODID, "worldTimer"), new WorldTimerProvider());
+        }
+
+        @SubscribeEvent
+        public void onTick(TickEvent.WorldTickEvent event)
+        {
+            if(event.world.isRemote)
+                return;
+
+            IWorldTimer timer = event.world.getCapability(TIMER_CAP, null);
+
+            if(event.phase == TickEvent.Phase.END)
+                timer.onTick();
+
+            if(event.phase == TickEvent.Phase.START){
+                if(!event.world.provider.getHasNoSky() && event.world.isRaining())
+                {
+                    if(!timer.timerIDs().contains("rainStopper"))
+                        timer.addTimer("rainStopper", 1000);
+                    else if(timer.isTimerReachedDuration("rainStopper")) {
+                        event.world.provider.resetRainAndThunder();
+                    }
+                } else
+                    timer.removeTimer("rainStopper");
+            }
+        }
+    }
+
+    public interface IWorldTimer {
```

Comment:
Since this is test mod for capability and there's nearly no hook in world to work with, I got this poor example. (though the timer is intended to be interacted from several mods)

So these possible usages would be more important:
- Celestial Dynamics (Days, Seasons - Behaviors of Sun, Moon, Stars)
- Atmosphere
- Environment
- Spawn Property
etc.